### DISCUSSION TITLE
Shortcut to generate a url to an item's own resource page

### DISCUSSION THREAD

**Discussion by edgrosvenor** [filamentphp/filament#213] [Ideas]
I have a resource for organizations and a resource for employees. They are related, so I use relationship managers to make it easy to access one from the other. But within the relationship managers, I want the tables and forms to have just high value, at a glance key data. Within the resources for each model, I want to be able to build out more complete tables and forms.

So say I'm looking at Organization 1 and I see the record for Employee A. I want to add a link to quickly jump to Employee A's record in the Employee Resource. I can do that by generating the URL in a function, which is fine. But this seems like a common enough use case that a shortcut might make sense. So instead of `->url( $callback )` I'd like to reach for something like `->resourceUrl() ` that just links to the record in its own model's resource (if it exists).

**Comment by ryangjchandler** [comment]
@edgrosvenor ~Interesting idea. I'll open a PR with something similar now.~ I won't promise this just yet, I need to make sure that we actually have everything available because the URL generator relies on the resource which the Relation Manager has no knowledge of.

**Reply by ryangjchandler** [reply]
Yeah, so the `RelationManager` doesn't actually know what `Resource` you're working with, it only cares about the model. So there's a couple of options really:

1. Add a new `resourceUrl` method and you can pass through the FQN of the `Resource` it should be using to generate the URL.
2. We add a new `$resource` property to the `RelationManager`.

Personally, the first one is my preference. It still requires an argument but it doesn't feel like as much hard work as the `url()` with a callback approach.

In the future, I'd like to suggest adding a `$resource` property so that we can automatically pull through the create/edit forms, removing the need to redefine them in the relation manager but for now, `resourceUrl()` should work:

```php
Columns\Text::make('name')
    ->resourceUrl(EmployeeResource::class)
```

What do you think @edgrosvenor?

**Reply by danharrin** [reply]
> Yeah, so the `RelationManager` doesn't actually know what `Resource` you're working with, it only cares about the model. So there's a couple of options really:
> 
> 
> 
> 1. Add a new `resourceUrl` method and you can pass through the FQN of the `Resource` it should be using to generate the URL.
> 
> 2. We add a new `$resource` property to the `RelationManager`.
> 
> 
> 
> Personally, the first one is my preference. It still requires an argument but it doesn't feel like as much hard work as the `url()` with a callback approach.
> 
> 
> 
> In the future, I'd like to suggest adding a `$resource` property so that we can automatically pull through the create/edit forms, removing the need to redefine them in the relation manager but for now, `resourceUrl()` should work:
> 
> 
> 
> ```php
> 
> Columns\Text::make('name')
> 
>     ->resourceUrl(EmployeeResource::class)
> 
> ```
> 
> 
> 
> What do you think @edgrosvenor?

We should have a second parameter to specify the name of the the route if it's not edit.

**Reply by ryangjchandler** [reply]
> > Yeah, so the `RelationManager` doesn't actually know what `Resource` you're working with, it only cares about the model. So there's a couple of options really:
> > 
> > 
> > 
> > 1. Add a new `resourceUrl` method and you can pass through the FQN of the `Resource` it should be using to generate the URL.
> > 
> > 2. We add a new `$resource` property to the `RelationManager`.
> > 
> > 
> > 
> > Personally, the first one is my preference. It still requires an argument but it doesn't feel like as much hard work as the `url()` with a callback approach.
> > 
> > 
> > 
> > In the future, I'd like to suggest adding a `$resource` property so that we can automatically pull through the create/edit forms, removing the need to redefine them in the relation manager but for now, `resourceUrl()` should work:
> > 
> > 
> > 
> > ```php
> > 
> > Columns\Text::make('name')
> > 
> >     ->resourceUrl(EmployeeResource::class)
> > 
> > ```
> > 
> > 
> > 
> > What do you think @edgrosvenor?
> 
> We should have a second parameter to specify the name of the the route if it's not edit.

Yes, agreed.

**Reply by danharrin** [reply]
> > > Yeah, so the `RelationManager` doesn't actually know what `Resource` you're working with, it only cares about the model. So there's a couple of options really:
> > > 
> > > 1. Add a new `resourceUrl` method and you can pass through the FQN of the `Resource` it should be using to generate the URL.
> > > 2. We add a new `$resource` property to the `RelationManager`.
> > > 
> > > Personally, the first one is my preference. It still requires an argument but it doesn't feel like as much hard work as the `url()` with a callback approach.
> > > In the future, I'd like to suggest adding a `$resource` property so that we can automatically pull through the create/edit forms, removing the need to redefine them in the relation manager but for now, `resourceUrl()` should work:
> > > ```
> > > Columns\Text::make('name')
> > > 
> > >     ->resourceUrl(EmployeeResource::class)
> > > ```
> > > 
> > > 
> > > What do you think @edgrosvenor?
> > 
> > 
> > We should have a second parameter to specify the name of the the route if it's not edit.
> 
> Yes, agreed.

These quote replies really do get a bit excessive after a while, don't they?

**Reply by edgrosvenor** [reply]
That would work for me.

**Comment by Daniel-H123** [comment]
Was this ever implemented?

**Reply by micraux** [reply]
Look [here](https://filamentphp.com/docs/3.x/panels/resources/getting-started#generating-urls-to-resource-pages)
